src/lib/Form/DataTransformer/RoleAssignmentTransformer.php

```diff
@@ -47,7 +49,7 @@ public function reverseTransform(mixed $value): ?APIRoleAssignment
             return null;
         }
 
-        if (!ctype_digit($value)) {
+        if (!is_int($value) && !ctype_digit($value)) {
```

Comment:
Reverse transformers now accept ints, but the thrown message still says “Expected a numeric string.” which is misleading for integer inputs. Consider updating the message (or narrowing accepted types) to keep validation feedback accurate.